frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -228,7 +228,12 @@ const getTopologyEdgeItems = (
             sbr.spec.backingServiceSelector.resourceRef &&
           _.get(deployment, 'metadata.ownerReferences[0].kind') ===
             sbr.spec.backingServiceSelector.kind;
-        return targetFound;
+        const appGroup = _.get(
+          deployment,
+          ['metadata', 'labels', 'app.kubernetes.io/part-of'],
+          null,
+        );
+        return !application ? targetFound : application === appGroup && targetFound;
```

Comment:
```
return targetFound && (!application || application === appGroup)
```
Probably do this since `targetFound` is required anyways so no need to add it twice.